commons-http/src/main/scala/com/avsystem/commons/http/okhttp/rest/RestClient.scala

```diff
@@ -0,0 +1,79 @@
+package com.avsystem.commons
+package http.okhttp.rest
+
+import java.io.IOException
+import java.net.URLEncoder
+
+import com.avsystem.commons.annotation.explicitGenerics
+import com.avsystem.commons.rest.RawRest.HandleRequest
+import com.avsystem.commons.rest.{HeaderValue, HttpBody, PathValue, QueryValue, RawRest, RestMetadata, RestResponse}
+import okhttp3.{Call, Callback, HttpUrl, MediaType, OkHttpClient, Request, RequestBody, Response}
+
+object RestClient {
+
+  def apply[@explicitGenerics Real: RawRest.AsRealRpc : RestMetadata](
+    client: OkHttpClient,
+    scheme: String,
```

Comment:
My intention in previous comment was to pass _single_ `baseUrl` which would contain protocol, host, port and path prefix. This would be analogous to Jetty-based `RestClient`. I think it's better because we have just one parameter instead of four and it's easy to build from parts while splitting the already built URL into scheme, host, port and path prefix is way more annoying.

Reply:
Yup, I know, but with OkHttp URL builder I can't pass baseUrl which contains protocol, host etc. I need to set all arguments separately. So I decided to expose this API instead of trying to split baseUrl into these parts.  Maybe the most elegant solution requires me to create a special case class for that or use something else, but I want to leave it as simple as possible.

Reply:
I think you can use `HttpUrl.parse(baseUrl).newBuilder` to obtain URL builder set up with base URL.